code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/process/HostRemovePostListener.java

```diff
@@ -21,10 +41,39 @@ public HandlerResult handle(ProcessState state, ProcessInstance process) {
         final Host host = (Host) state.getResource();
         CacheManager cm = CacheManager.getCacheManagerInstance(this.objectManager);
         cm.removeHostInfo(host.getId());
+        removeHostCacheFromScheduler(host.getId(), host.getAccountId());
 
         return null;
     }
 
+
+    @Inject
+    IdFormatter idFormatter;
+
+    public void removeHostCacheFromScheduler(Long hostId, Long envId) {
+        String ALLOCATE_CPU_MEMORY_URL = SCHEDULER_URL + "/remove-host";
+        List<BasicNameValuePair> requestData = new ArrayList<>();
+
+        requestData.add(new BasicNameValuePair("hostId", (String) idFormatter.formatId(HostConstants.TYPE, hostId)));
+        requestData.add(new BasicNameValuePair("envId", (String) idFormatter.formatId(AccountConstants.TYPE, envId)));
+
+        HttpResponse response;
+        try {
+            response = Request.Post(ALLOCATE_CPU_MEMORY_URL).addHeader("Accept", "application/json")
+                    .bodyForm(requestData).execute().returnResponse();
+            int statusCode = response.getStatusLine().getStatusCode();
+            if (statusCode != 200) {
+                log.error("statusCode: {}", statusCode);
+            }
+
+        } catch (HttpHostConnectException ex) {
+            log.error("Scheduler Service not reachable at [{}]", ALLOCATE_CPU_MEMORY_URL);
+        } catch (IOException e) {
+            log.error((e.getStackTrace()).toString(), e);
+        }
+
+        return;
```

Comment:
Again, when status code != 200 or an exception thrown, should we error out here? Does the microservice return any particular error code when host no longer exists for the environment? if so, we should just ignore that error code, and for all the rest handle the exception